### PR TITLE
[SHELL32][PAINT][DRIVERS][SHELLEXT] French translation update

### PR DESCRIPTION
**Pull request by KyleKatarn31**
## Purpose

French Translation following
- https://github.com/reactos/reactos/pull/3056
- https://github.com/reactos/reactos/pull/3100

Previous updates on Paint
Missing FR translation in drivers

### CODE REVIEW

**Comment by binarymaster** [comment]
Cc @reactos/lang-french

**Comment by KyleKatarn31** [comment]
Sorry ColinFinck, Extravert-ir, HeisSpiter, ThFabba and tkreuzer for the undue review request (wrong Git command)

**Comment by KyleKatarn31** [comment]
@HBelusca @sdever would you please review and approve if a french native speaker is needed before go to merge ?

**Comment by KyleKatarn31** [comment]
@katahiromz @binarymaster : Do you need anything from me in order to Approve & merge ?
(like for https://github.com/reactos/reactos/pull/3047 even if RH does not apply here)

**Review by katahiromz — APPROVED** [review]

**Review by sdever — APPROVED** [review]

File: dll/shellext/stobject/lang/fr-FR.rc [hunk 1]
```diff
@@ -16,26 +16,26 @@ STRINGTABLE
...
+    IDS_PWR_METER             "&Ouvrir le gestionnaire d'énergie"
     IDS_PWR_RUN               "shell32.dll,Control_RunDLL PowerCfg.cpl"
-    IDS_PWR_PERCENT_REMAINING "%u%% remaining"
-    IDS_PWR_CHARGING          "%u%% and charging"
-    IDS_PWR_UNKNOWN_REMAINING "Unknown remaining"
-    IDS_PWR_AC "On AC power"
-    IDS_PWR_HOURS_REMAINING "%1!u!:%2!02u! hours (%3!u!%%) remaining"
-    IDS_PWR_MINUTES_REMAINING "%1!u! min (%2!u!%%) remaining"
+    IDS_PWR_PERCENT_REMAINING "%u%% restants"
+    IDS_PWR_CHARGING          "%u%% et en charge"
```
Comment (HBelusca) [inline dll/shellext/stobject/lang/fr-FR.rc:22]:
```suggestion
    IDS_PWR_CHARGING          "%u%% en charge"
```

File: dll/shellext/stobject/lang/fr-FR.rc [hunk 2]
```diff
@@ -16,26 +16,26 @@ STRINGTABLE
...
+    IDS_PWR_UNKNOWN_REMAINING "Reste inconnu"
```
Comment (HBelusca) [inline dll/shellext/stobject/lang/fr-FR.rc:23]:
```suggestion
    IDS_PWR_UNKNOWN_REMAINING "Restant inconnu"
```

File: dll/shellext/stobject/lang/fr-FR.rc [hunk 3]
```diff
@@ -16,26 +16,26 @@ STRINGTABLE
...
     IDS_HOTPLUG_COMMA ", "
-    IDS_HOTPLUG_DRIVE " - Drive(%s)"
-    IDS_HOTPLUG_DRIVES " - Drives(%s)"
+    IDS_HOTPLUG_DRIVE " - Lecteur(%s)"
+    IDS_HOTPLUG_DRIVES " - Lecteurs(%s)"
     IDS_HOTPLUG_A "A:"
-    IDS_HOTPLUG_REQUIRES "This hardware requires ""Safe Removal"""
-    IDS_HOTPLUG_CLICK "Before removing this hardware, click on this icon and select the hardware you want to remove."
+    IDS_HOTPLUG_REQUIRES "Ce périphérique nécessite un ""retrait en sécurité"""
+    IDS_HOTPLUG_CLICK "Avant de retirer ce matériel, cliquez sur cet icone et sélectionnez le matériel à retirer."
```
Comment (HBelusca) [inline dll/shellext/stobject/lang/fr-FR.rc:38]:
```suggestion
    IDS_HOTPLUG_CLICK "Avant de retirer ce périphérique, cliquez sur cette icône et sélectionnez le périphérique à retirer."
```

File: dll/win32/shell32/lang/fr-FR.rc [hunk 4]
```diff
@@ -998,12 +998,12 @@ BEGIN
...
-    IDS_COPYTOMENU "Copy to &folder..."
-    IDS_COPYTOTITLE "Select the place where you want to copy '%s'. Then click the Copy button."
-    IDS_COPYITEMS "Copy Items"
-    IDS_COPYBUTTON "Copy"
-    IDS_MOVETOMENU "Mo&ve to folder..."
-    IDS_MOVETOTITLE "Select the place where you want to move '%s' to. Then click the Move button."
-    IDS_MOVEITEMS "Move Items"
-    IDS_MOVEBUTTON "Move"
+    IDS_COPYTOMENU "Copier vers le &dossier..."
+    IDS_COPYTOTITLE "Selectionnez l'emplacement où vous souhaitez copier '%s' puis appuyez sur le bouton Copier."
```
Comment (HBelusca) [inline dll/win32/shell32/lang/fr-FR.rc:1002]:
```suggestion
    IDS_COPYTOTITLE "Sélectionnez l'emplacement où vous souhaitez copier '%s', puis appuyez sur le bouton Copier."
```

File: dll/win32/shell32/lang/fr-FR.rc [hunk 5]
```diff
@@ -998,12 +998,12 @@ BEGIN
...
+    IDS_COPYITEMS "Copier les éléments"
+    IDS_COPYBUTTON "Copier"
+    IDS_MOVETOMENU "&Déplacer vers le dossier..."
+    IDS_MOVETOTITLE "Selectionnez l'emplacement où vous souhaitez déplacer '%s' puis appuyez sur le bouton Déplacer."
```
Comment (HBelusca) [inline dll/win32/shell32/lang/fr-FR.rc:1006]:
```suggestion
    IDS_MOVETOTITLE "Sélectionnez l'emplacement où vous souhaitez déplacer '%s', puis appuyez sur le bouton Déplacer."
```

File: drivers/storage/class/cdrom/cdrom.inf [hunk 6]
```diff
@@ -57,6 +57,13 @@ CDROMClassName = "DVD/CD-ROM drives"
 GenCdRom.DeviceDesc = "CD-ROM Drive"
 GenericMfg = "(Generic DVD/CD-ROM Drives)"
 
+[Strings.040C]
+ReactOS = "Equipe ReactOS"
+CDROMClassName = "Lecteurs DVD/CD-ROM"
+GenCdRom.DeviceDesc = "Lecteur CD-ROM"
+GenericMfg = "(Lecteurs DVD/CD-ROM génériques)"
+
+
```
Comment (HBelusca) [inline drivers/storage/class/cdrom/cdrom.inf:66]:
```suggestion
```

File: dll/shellext/stobject/lang/fr-FR.rc [hunk 7]
```diff
@@ -16,26 +16,26 @@ STRINGTABLE
...
+    IDS_PWR_AC "Sur secteur"
+    IDS_PWR_HOURS_REMAINING "%1!u!:%2!02u! heures (%3!u!%%) restantes"
+    IDS_PWR_MINUTES_REMAINING "%1!u! mins (%2!u!%%) restantes"
 
     //Hotplug related strings
-    IDS_HOTPLUG_REMOVE_1 "Safely Remove Hardware"
-    IDS_HOTPLUG_REMOVE_2 "&Safely Remove Hardware"
-    IDS_HOTPLUG_REMOVE_3 "Safely remove %s"
+    IDS_HOTPLUG_REMOVE_1 "Retirer le matériel en sécurité"
+    IDS_HOTPLUG_REMOVE_2 "&Retirer le matériel en sécurité"
```
Comment (HBelusca) [inline dll/shellext/stobject/lang/fr-FR.rc:30]: Maybe s/matériel/périphérique/ (as done below)

Comment (KyleKatarn31) [reply dll/shellext/stobject/lang/fr-FR.rc:30]: I align with "Périphérique" in all places for the sake of consistency with your other comments.

**Comment by KyleKatarn31** [comment]
All comments from @HBelusca taken into account (thanks !)

File: drivers/storage/class/cdrom/cdrom.inf [hunk 8]
```diff
@@ -57,6 +57,12 @@ CDROMClassName = "DVD/CD-ROM drives"
 GenCdRom.DeviceDesc = "CD-ROM Drive"
 GenericMfg = "(Generic DVD/CD-ROM Drives)"
 
+[Strings.040C]
```
Comment (ThFabba) [inline drivers/storage/class/cdrom/cdrom.inf:60]: Can you keep the sections sorted please? Ignore the '04', then it goes 08,09,0a,0b,0c, ...
And in doing that, you'd see there is already a .040c section.

File: drivers/storage/class/ramdisk/ramdisk.inf [hunk 9]
```diff
@@ -74,3 +74,10 @@ RamdiskClassName = "RAM-диски"
 RamdiskBus.Desc = "Контроллер RAM-диска"
 RamdiskVolume.Desc = "Устройство RAM-диска (том)"
 Ramdisk.ServiceDesc = "Драйвер RAM-диска ReactOS"
+
+[Strings.040C]
```
Comment (ThFabba) [inline drivers/storage/class/ramdisk/ramdisk.inf:78]: This should go before 0415.

File: drivers/storage/class/disk/disk.inf [hunk 10]
```diff
@@ -66,6 +66,14 @@ Disk.DeviceDesc = "Disk Drive"
 OpticalDisk.DeviceDesc = "Optical Disk Drive"
 Disk.ServiceDesc = "Disk Driver"
 
+[Strings.040C]
```
Comment (ThFabba) [inline drivers/storage/class/disk/disk.inf:69]: Same here, move down please.

File: drivers/storage/class/cdrom/cdrom.inf (hunk 8 above)
Comment (KyleKatarn31) [reply drivers/storage/class/cdrom/cdrom.inf:60]: ok

File: drivers/storage/class/ramdisk/ramdisk.inf (hunk 9 above)
Comment (KyleKatarn31) [reply drivers/storage/class/ramdisk/ramdisk.inf:78]: ok

File: drivers/storage/class/disk/disk.inf (hunk 10 above)
Comment (KyleKatarn31) [reply drivers/storage/class/disk/disk.inf:69]: ok

**Comment by KyleKatarn31** [comment]
Comments from @ThFabba taken into account. Ready to go ?

**Comment by KyleKatarn31** [comment]
Would it be possible to merge now that no more comments are open ?

File: dll/win32/shell32/lang/fr-FR.rc [hunk 11]
```diff
@@ -998,12 +998,12 @@ BEGIN
...
     IDS_LINK_INVALID "L'élément '%s' auquel ce raccourci fait référence a été modifié ou déplacé et ce raccourci ne fonctionne plus correctement."
-    IDS_COPYTOMENU "Copy to &folder..."
-    IDS_COPYTOTITLE "Select the place where you want to copy '%s'. Then click the Copy button."
-    IDS_COPYITEMS "Copy Items"
-    IDS_COPYBUTTON "Copy"
-    IDS_MOVETOMENU "Mo&ve to folder..."
-    IDS_MOVETOTITLE "Select the place where you want to move '%s' to. Then click the Move button."
-    IDS_MOVEITEMS "Move Items"
-    IDS_MOVEBUTTON "Move"
+    IDS_COPYTOMENU "Copier vers le &dossier..."
```
Comment (HBelusca) [inline dll/win32/shell32/lang/fr-FR.rc:1001]:
```suggestion
    IDS_COPYTOMENU "Copie&r dans un dossier..."
```

File: dll/win32/shell32/lang/fr-FR.rc [hunk 12]
```diff
@@ -998,12 +998,12 @@ BEGIN
...
+    IDS_COPYTOTITLE "Sélectionnez l'emplacement où vous souhaitez copier '%s', puis appuyez sur le bouton Copier."
+    IDS_COPYITEMS "Copier les éléments"
+    IDS_COPYBUTTON "Copier"
+    IDS_MOVETOMENU "&Déplacer vers le dossier..."
```
Comment (HBelusca) [inline dll/win32/shell32/lang/fr-FR.rc:1005]:
```suggestion
    IDS_MOVETOMENU "&Déplacer vers un dossier..."
```

**Review by HBelusca — APPROVED** [review]
Fix keyboard accelerators.

**Comment by KyleKatarn31** [comment]
Thank you @HBelusca

**Pull request merged**